trl/trainer/grpo_trainer.py

```diff
@@ -904,36 +904,63 @@ def _generate_and_score_completions(
         self._metrics[mode]["reward"].append(mean_grouped_rewards.mean().item())
         self._metrics[mode]["reward_std"].append(std_grouped_rewards.mean().item())
 
+        # Log completions when we complete a full gradient accumulation cycle
+        # For logging across the gradient accumulation steps, we need to accumulate the data
+        is_last_step_in_grad_accum = (
+            self._step % self.args.gradient_accumulation_steps == self.args.gradient_accumulation_steps - 1
+        )
+        should_log_completions = (
+            self.log_completions
+            and self.state.global_step % self.args.logging_steps == 0
+            and is_last_step_in_grad_accum
+        )
+
+        # Collect data for logging throughout the accumulation steps
```

Comment:
I think `self.control.should_log` achieves the same goal. Let me check

Reply:
So yes it achieves the same goal but this flag is only update after the `loss.backward` so we can't use it here. I'm starting to think that it would make more sense to have all this code in `self.log`. I'm checking if it's possible.

Reply:
So yes It's possible, I'll do it in a follow-up PR